### PR TITLE
Added input fields to the employment history section

### PR DESCRIPTION
**Pull request by BenVuong**

### CODE REVIEW

**Review by MatthewBackes — DISMISSED** [review]

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 1]
```diff
@@ -1,7 +1,257 @@
...
+import FormControl from "@mui/material/FormControl";
+import Grid from "@mui/material/Grid";
+import InputField from "../../../components/inputs/InputField";
+import TextField from "../../../components/inputs/TextField";
+import Box from "@mui/material/Box";
 
 function EmploymentHistory() {
-  return <></>;
+  const [company, setCompany] = useState("");
+  const [comapanyAddress, setCompanyAddress] = useState("");
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:10]: *companyAddress

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 2]
```diff
@@ -1,7 +1,257 @@
...
+  const [jobTitle, setJobIitle] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [reason, setReason] = useState("");
+
+  const [companyHelper, setCompanyHelper] = useState("");
+  const [companyAddressHelper, setCompanyAddressHelper] = useState("");
+  const [jobTitleHelper, setJobIitleHelper] = useState("");
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:18]: *setJobTitleHelper

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 3]
```diff
@@ -1,7 +1,257 @@
...
+  const [startDateHelper, setStartDateHelper] = useState("");
+  const [endDateHelper, setEndDateHelper] = useState("");
+  const [reasonHelper, setReasonHelper] = useState("");
+
+  const [companyError, setComapnyError] = useState(false);
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:23]: *setCompanyError

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 4]
```diff
@@ -1,7 +1,257 @@
...
+  const [startDateError, setStartDateError] = useState(false);
+  const [endDateError, setEndDateError] = useState(false);
+  const [reasonError, setReasonError] = useState(false);
+
+  function handleSaveInformation(e) {
+    e.preventDefault();
+    // console.log("TODO: Submit data to backend!");
+  }
+
+  function handleChnage(event) {
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:35]: *handleChange(event)

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 5]
```diff
@@ -1,7 +1,257 @@
...
+    const e = event;
+    const val = e.target.value;
+    const name = e.target.name;
+
+    if (name === "company") {
+      setCompany(val);
+      if (val.length > 0) {
+        setComapnyError(false);
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:43]: *setCompanyError(false)

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 6]
```diff
@@ -1,7 +1,257 @@
...
+        setCompanyHelper("");
+      } else {
+        setComapnyError(true);
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:46]: setCompanyError(true);

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 7]
```diff
@@ -1,7 +1,257 @@
...
+        setCompanyAddressError(false);
+        setCompanyAddressHelper("");
+      } else {
+        setCompanyAddressError(true);
+        setCompanyAddressHelper(
+          "Please enter in the address of the company you previously worked for"
+        );
+      }
+    } else if (name === "jobTitle") {
+      setJobIitle(val);
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:63]: setJobTitle(val);

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 8]
```diff
@@ -1,7 +1,257 @@
...
+      if (val.length > 0) {
+        setJobTitleError(false);
+        setJobIitleHelper("");
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:66]: setJobTitleHelper("");

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 9]
```diff
@@ -1,7 +1,257 @@
...
+      } else {
+        setJobTitleError(true);
+        setJobIitleHelper(
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:69]: setJobTitleHelper

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 10]
```diff
@@ -1,7 +1,257 @@
...
+                maxWidth: "500px",
+              }}
+            >
+              <InputField
+                title={"Company"}
+                id="company"
+                value={company}
+                autoFocus={true}
+                error={companyError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:134]: *handeChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 11]
```diff
@@ -1,7 +1,257 @@
...
+                marginBottom: 1,
+                marginLeft: 3,
+                width: "90%",
+                maxWidth: "500px",
+              }}
+            >
+              <InputField
+                title={"Company Address"}
+                id="companyAddress"
+                value={comapanyAddress}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:154]: *companyAddress

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 12]
```diff
@@ -1,7 +1,257 @@
...
+                error={companyAddressError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:156]: *handleChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 13]
```diff
@@ -1,7 +1,257 @@
...
+                width: "90%",
+                maxWidth: "500px",
+              }}
+            >
+              <InputField
+                title={"Job Title"}
+                id="jobTitle"
+                value={jobTitle}
+                error={jobTitleError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:178]: *handleChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 14]
```diff
@@ -1,7 +1,257 @@
...
+                width: "90%",
+                maxWidth: "500px",
+              }}
+            >
+              <InputField
+                title={"Start Date"}
+                id="startDate"
+                value={startDate}
+                error={startDateError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:200]: *handleChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 15]
```diff
@@ -1,7 +1,257 @@
...
+                width: "90%",
+                maxWidth: "500px",
+              }}
+            >
+              <InputField
+                title={"End Date"}
+                id="endDate"
+                value={endDate}
+                error={endDateError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:222]: *handleChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 16]
```diff
@@ -1,7 +1,257 @@
...
+                width: "90%",
+                maxWidth: "500px",
+              }}
+            >
+              <TextField
+                title={"Reason For Leaving"}
+                id="reason"
+                value={reason}
+                error={reasonError}
+                onChangeHandler={handleChnage}
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:244]: *handleChange

File: onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx [hunk 17]
```diff
@@ -1,7 +1,257 @@
...
+import Grid from "@mui/material/Grid";
+import InputField from "../../../components/inputs/InputField";
+import TextField from "../../../components/inputs/TextField";
+import Box from "@mui/material/Box";
 
 function EmploymentHistory() {
-  return <></>;
+  const [company, setCompany] = useState("");
+  const [comapanyAddress, setCompanyAddress] = useState("");
+  const [jobTitle, setJobIitle] = useState("");
```
Comment (MisheelSodgerel) [inline onboard/src/pages/subsections/backgroundInformation/EmploymentHistory.jsx:11]: *setJobTitle

**Review by MisheelSodgerel — CHANGES_REQUESTED** [review]
added comments about spelling

**Review by MisheelSodgerel — APPROVED** [review]

**Review by MisheelSodgerel — APPROVED** [review]

**Review by MatthewBackes — APPROVED** [review]

**Pull request merged**